PlexRequests.UI/Modules/AdminModule.cs

```diff
@@ -470,6 +471,19 @@ private Response GetSonarrQualityProfiles()
             return Response.AsJson(profiles);
         }
 
+        private Response GetSonarrRootFolders()
+        {
+            var settings = this.Bind<SonarrSettings>();
```

Comment:
Instead of doing the validation in the Javascript, you can check if the binding is valid. There are a bunch of validators e.g. SonarrSettingsValidator (If you look at some of the POST requests you should be able to see how we validate, and when there is invalid data we return the invalid message)